content/blog/using-docker-wrong-my-journey-to-a-better-container.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Using Docker Wrong: My Journey to a Better Container"
+date: 2021-07-13T06:03:11.023Z
+author: John Omernik
+authorimage: /img/Avatar1.svg
+---
+**Editor’s Note:** MapR products and solutions sold prior to the acquisition of such assets by Hewlett Packard Enterprise Company in 2019 may have older product names and model numbers that differ from current solutions. For information about current offerings, which are now part of HPE Ezmeral Data Fabric, please visit [https://www.hpe.com/us/en/software/data-fabric.html](https://www.hpe.com/us/en/software/data-fabric.html)
+
+## Original Post Information:
+
+```markdown
+"authorDisplayName": "John Omernik",
+"publish": "2018-06-07T13:00:00.000",
+"category": "use-case",
+```
+---
+
+Since starting at MapR, I’ve been focusing on a number of things, including learning more about MapR, meeting coworkers and customers, and a topic near and dear to my heart: immersing myself in containers and integration with the MapR Data Platform.  In past jobs, I’ve used MapR with containers but have not dug into the Persistent Application Client Container (PACC) nearly as much as I should.  This cautionary tale is one to remind everyone how amazing containers are AND how to keep the benefits of containers at the top of your mind when working to create them, so you don’t design your containers in such a way that you give up those benefits and end up writing a mea culpa blog like this one.  
+
+![](/img/container-wide.jpg)
+
+I was tasked to come up with a way to have a syslog server receive log messages and produce those messages to MapR Event Store – a powerful setup for Information Security and IT Operations professions.  I had a few requirements:
+
+1. Must run in Docker.  This is now a self-imposed philosophy for all things.  I want everything to be portable, sharable, and usable by everyone on the MapR Data Platform.
+2. Simple in and out.  It has to listen on a syslog port (TCP and UDP) and produce to MapR Event Store.  I wanted to show how it works in the Dockerfile, but I didn’t want to chain multiple containers together.
+
+After some research I decided to utilize a great MapR partner, [StreamSets](https://streamsets.com/), and their [StreamSets Data Collector](https://streamsets.com/products/sdc) (SDC) as the core.  The SDC’s interface, capabilities, and integrations with MapR would serve me well here.  I had worked with StreamSets before in Docker with MapR; however, I had NOT used the MapR PACC, so it would be a learning experience.  I pulled out my old Dockerfiles, modified some components, ensured I was handling my newly secured cluster with tickets, and it worked great!  I even produced a script to create the MapR Event Store and generate test messages, all contained in one handy GitHub repository.
+
```

Comment:
You may want to "hotlink" the Github URL, so it will read as:
...all contained in one handy [GitHub repository](https://github.com/johnomernik/maprssss_wrong)